SharpPcap/WinDivert/WinDivertDevice.cs

```diff
@@ -191,13 +215,56 @@ public void SendPacket(byte[] p, int size)
 
         public void SendPacket(ReadOnlySpan<byte> p)
         {
-            // TODO
-            throw new NotSupportedException();
+            SendPacket(p, GetAddress(p));
+        }
+
+        private void SendPacket(ReadOnlySpan<byte> p, WinDivertAddress addr)
+        {
+            ThrowIfNotOpen();
+            bool res;
+            unsafe
+            {
+                fixed (byte* p_packet = p)
+                {
+                    res = WinDivertNative.WinDivertSend(Handle, new IntPtr(p_packet), (uint)p.Length, out var pSendLen, ref addr);
+                }
+            }
+            if (!res)
+            {
+                ThrowLastWin32Error("Can't send packet");
+            }
+        }
+
+        private static WinDivertAddress GetAddress(ReadOnlySpan<byte> p)
+        {
+            var version = p[0] >> 4;
+            IPAddress src;
+            IPAddress dst;
+            if (version == 4)
+            {
+                src = new IPAddress(p.Slice(12, 4).ToArray());
```

Comment:
can we point to where the 12, 16, 8, 24 numbers come from in case someone is looking later?